src/Datadog.Trace.ClrProfiler.Native/cor_profiler.cpp

```diff
@@ -164,6 +169,40 @@ HRESULT STDMETHODCALLTYPE CorProfiler::ModuleLoadFinished(ModuleID module_id,
     return S_OK;
   }
 
+  if (debug_logging_enabled) {
+    Debug("ModuleLoadFinished: ", module_id, " ", module_info.assembly.name,
+            " AppDomain ", module_info.assembly.app_domain_id, " ",
+            module_info.assembly.app_domain_name);
+  }
+
+  AppDomainID app_domain_id = module_info.assembly.app_domain_id;
+
+  // Identify the AppDomain ID of mscorlib which will be the Shared Domain
+  // because mscorlib is always a domain-neutral assembly
+  if (!mscorlib_module_loaded && module_info.assembly.name == "mscorlib"_W) {
```

Comment:
Can we just assume the first assembly is what we need for this? Do we need the mscorlib name?

Reply:
I don't think it hurts to add this extra check. It seems the only time that assumption was violated was loading `System.Private.CoreLib` when running on CoreCLR

Reply:
Would apps targeting `netstandard` run into that as well?

Reply:
I'm not sure what you mean. The runtime startup behavior is only dependent on whether .NET Framework or .NET Core is being started.

Reply:
Ah, for some reason I thought netstandard was another equivalent. 🤔 
Would it be reasonable to allow this if it's `mscorlib` or `System.Private.CoreLib`?
Less to think about later?

Reply:
And I guess the reason I would have opted for just assuming the first module load finished was the entry module is that it would save us from any whacky scenarios in the future, where the entry is validly not mscorlib. I'm comfortable with whatever you choose though.

Reply:
I'll add the check against `System.Private.CoreLib` to get the first assembly in the .NET Core case 👍